completion/models.py

```diff
@@ -168,6 +168,38 @@ class BlockCompletion(TimeStampedModel, models.Model):
 
     objects = BlockCompletionManager()
 
+    @classmethod
+    def get_course_completions(cls, user, course_key):
+        """
+        query all completions for course/user pair
+
+        Return value:
+            dict[BlockKey] = float
+        """
+        course_block_completions = cls.objects.filter(
+            user=user,
+            course_key=course_key,
+        )
+        # will not return if <= 0.0
+        return {completion.block_key: completion.completion for completion in course_block_completions}
+
+    @classmethod
+    def get_latest_block_completed(cls, user, course_key):
+        """
+        query latest completion for course/user pair
```

Comment:
You can squash this string all into one sentence: "Returns the most recently modified BlockCompletion object for the given user and course_key, or None if no such BlockCompletion exists."

Reply:
💯